app/services/notification/ScheduledNotificationRunner.scala

```diff
@@ -1,22 +1,33 @@
 package services.notification
 
+import javax.inject.{Inject, Singleton}
+
 import com.amazonaws.services.simpleemail.model._
-import config.AMIableConfig
+import config.{AMIableConfig, AmiableConfigProvider}
 import models._
+import play.api.{Environment, Logger, Mode}
 import prism.{Prism, PrismLogic}
 
 import scala.concurrent.ExecutionContext
 
-object ScheduledNotificationRunner {
-  def run(mailClient: AWSMailClient)(implicit config: AMIableConfig, ec: ExecutionContext): Attempt[List[String]] = {
+@Singleton
+class ScheduledNotificationRunner@Inject()(mailClient: AWSMailClient, environment: Environment, amiableConfigProvider: AmiableConfigProvider)(implicit ec: ExecutionContext) {
+  implicit val config = amiableConfigProvider.conf
+
+  def run(): Attempt[List[String]] = {
     for {
       instancesWithAmis <- Prism.instancesWithAmis(SSA(stage = Some("PROD")))
       oldInstances = PrismLogic.oldInstances(instancesWithAmis)
       ownersWithDefault <- Prism.getOwners
       ownersAndOldInstances = findInstanceOwners(oldInstances, ownersWithDefault)
       mailIds <- Attempt.traverse(ownersAndOldInstances.toList) { case (owner, oldInstancesForOwner) =>
-        val request = createEmailRequest(owner, oldInstancesForOwner)
-        mailClient.send(owner, request)
+        val instances = oldInstancesForOwner.sortBy(i => (i.stack, i.stage, i.app.headOption))
+        val request = createEmailRequest(owner, instances, config)
+        if(environment.mode == Mode.Prod || config.overrideToAddress.nonEmpty)
```

Comment:
Or... how does something like this feel? It moves all of the development / override logic into a single location. It would be ideal if this could be extracted into the companion object and have some test coverage (unfortunately I think you might need to mock the mail client).

```scala
(environment.mode, config.overrideToAddress) match {
  case (Mode.Prod, maybeOverride) =>
    mailClient.send(maybeOverride.getOrElse(s"${owner.id}@guardian.co.uk"), owner, request)
  case (_, Some(override)) =>
    mailClient.send(override, owner, request)
  case (_, None) =>
    Logger.info(s"Not in Prod and no override To Address set. Would have sent email to ${owner.id}, request: $request")
    Attempt.Right("")
}
```

Maybe @adamnfish has some better ideas on how to do this.